src/renderer/components/+preferences/preferences.tsx

```diff
@@ -33,178 +27,101 @@ export class Preferences extends React.Component {
+            <section>
+              <h1>Application</h1>
+            </section>
+            <section id="appearance">
+              <h2>Appearance</h2>
+              <SubTitle title="Theme"/>
+              <Select
+                options={this.themeOptions}
+                value={preferences.colorTheme}
+                onChange={({ value }: SelectOption) => preferences.colorTheme = value}
+              />
+            </section>
+            <section id="proxy">
+              <h2>Proxy</h2>
+              <SubTitle title="HTTP Proxy"/>
+              <Input
+                theme="round-black"
+                placeholder="Type HTTP proxy url (example: http://proxy.acme.org:8080)"
+                value={this.httpProxy}
+                onChange={v => this.httpProxy = v}
+                onBlur={() => preferences.httpsProxy = this.httpProxy}
+              />
+              <small className="hint">
+                Proxy is used only for non-cluster communication.
+              </small>
+
+              <SubTitle title="Certificate Trust"/>
+              <Checkbox
+                label="Allow untrusted Certificate Authorities"
+                value={preferences.allowUntrustedCAs}
+                onChange={v => preferences.allowUntrustedCAs = v}
+              />
+              <small className="hint">
+                This will make Lens to trust ANY certificate authority without any validations.{" "}
+                Needed with some corporate proxies that do certificate re-writing.{" "}
+                Does not affect cluster communications!
+              </small>
+            </section>
+            <section id="startup">
+              <h2>Start-up</h2>
+              <SubTitle title="Automatic Start-up"/>
+              <Checkbox
+                label="Automatically start Lens on login"
+                value={preferences.openAtLogin}
+                onChange={v => preferences.openAtLogin = v}
+              />
+            </section>
+          </section>
+
+          <section id="kubernetes">
+            <section>
+              <h1>Kubernetes</h1>
+            </section>
+            <section id="kubectl">
+              <h2>Kubectl binary</h2>
+              <KubectlBinaries preferences={preferences}/>
+            </section>
+            <section id="helm">
+              <h2>Helm Charts</h2>
+              <HelmCharts/>
+            </section>
+          </section>
+
+          <section id="extensions">
+            <section>
+              <h1>Extensions</h1>
+            </section>
+            {appPreferenceRegistry.getItems().map(({ title, components: { Hint, Input } }, index) => {
+              return (
+                <section key={index} id={title}>
+                  <h2>{title}</h2>
+                  <Input/>
+                  <small className="hint">
+                    <Hint/>
+                  </small>
+                </section>
+              );
+            })}
```

Comment:
This assumes that every extension only has one preference. I think it would be better if each extension had its own tree.

Reply:
I was looking at https://user-images.githubusercontent.com/1446224/105025079-82510580-5a55-11eb-92c4-cba13a4a8217.png

Every extension has it's own `section` element along with a link in navigation bar.

Reply:
How so? Currently none of the built in extensions have more than one but we should support it. This does mean that `appPreferenceRegistry` needs to be updated to keep track of that. Or even better you could do something like I propose in #2245 but that is probably out of scope of this PR.